File: src/navigation.ts
import HomeIcon from "@components/Icons/HomeIcon.astro";
import PostIcon from "@components/Icons/PostIcon.astro";
import PersonIcon from "@components/Icons/PersonIcon.astro";

export const navigation = {
    navigation: [
        {
            name: "Index",
            href: "/",
            icon: HomeIcon,
        },
        {
            name: "Posts",
            href: "/posts",
            icon: PostIcon,
        },
        {
            name: "About",
            href: "/about",
            icon: PersonIcon,
        }
    ]
};

File: Sources/Website/Navigation.swift
import Web

struct NavigationItem {
    let name: String
    let href: String
    let icon: Component
}

let navigation = [
    NavigationItem(
        name: "Index",
        href: "/",
        icon: HomeIcon()
    ),
    NavigationItem(
        name: "Posts",
        href: "/posts",
        icon: PostIcon()
    ),
    NavigationItem(
        name: "About",
        href: "/about",
        icon: PersonIcon()
    )
]
